internal/controllers/addon/utils.go

```diff
@@ -238,3 +220,53 @@ func (r *AddonReconciler) parseAddonInstallConfig(
 	}
 	return targetNamespace, catalogSourceImage, false
 }
+
+// HasMonitoringFederation is a helper to determine if a given addon's spec
+// defines a Monitoring.Federation.
+func HasMonitoringFederation(addon *addonsv1alpha1.Addon) bool {
```

Comment:
All these functions here can be made private, as they are not used in any other package.

Reply:
So I agree these are private helpers, but a few of them were being used by the integration suite ([example](https://github.com/openshift/addon-operator/blob/fadf3af25b54eb5950518023b5a3589d0af8d0ca/integration/monitoring_federation_test.go#L78)). IMO sharing these functions kind  of defeats the purpose of integration so I'm happy to make them private and replace the integration instances with fixtures. Let me know what you'd prefer and I will update.

Reply:
It's alright, nothing to stall this PR over, didn't see that it's used from the integration tests